src/MySqlConnector/MySqlClient/MySqlConnection.cs

```diff
@@ -198,6 +200,34 @@ internal MySqlSession Session
 			}
 		}
 
+		internal async Task<CachedProcedure> GetCachedProcedure(IOBehavior ioBehavior, string name, CancellationToken cancellationToken)
+		{
+			if (State != ConnectionState.Open)
+				throw new InvalidOperationException("Connection is not open.");
+			if (!m_cachedProceduresSupported)
+				return null;
+			if (m_cachedProcedures == null)
+				m_cachedProcedures = new Dictionary<string, CachedProcedure>();
+
+			var normalized = NormalizedSchema.MustNormalize(name);
+			CachedProcedure cachedProcedure;
+			if (m_cachedProcedures.TryGetValue(normalized.CacheKey, out cachedProcedure))
+				return cachedProcedure;
+			try
+			{
+				cachedProcedure = await CachedProcedure.FillAsync(ioBehavior, this, normalized.Schema, normalized.Component, cancellationToken).ConfigureAwait(false);
+				m_cachedProcedures[normalized.CacheKey] = cachedProcedure;
+			}
+			catch (MySqlException ex)
+			{
+				if (ex.Number != 1109)
+					throw;
+				// 1109 means information_schema.parameters table not found on MySql < 5.5.3, see http://dev.mysql.com/doc/refman/5.5/en/parameters-table.html
+				m_cachedProceduresSupported = false;
```

Comment:
Opting for an Exception Error Number Check here instead of checking the Server version before the query.  This way we don't have to maintain a list of server types and versions that support `information_schema.parameters` (MySql >= 5.5.3, MariaDB >= whatever, etc).  Is this acceptable?